examples/wrapper-types/src/wrapper-types.udl

```diff
@@ -4,9 +4,27 @@ typedef string JsonObject;
 [Wrapped]
 typedef i64 Handle;
 
+// Dictionary that we want to wrap in the foreign bindings code with a flat version
+dictionary UserAccount {
+    UserInfo info;
+    AccountInfo account;
+};
+
+dictionary UserInfo {
+    string name;
+};
+
+dictionary AccountInfo {
+    i32 id;
+};
+
+[Wrapped]
+typedef UserAccount WrappedUserAccount;
```

Comment:
I can't think of a good example that wraps timestamp.  How would you do it?

IMO, one thing that makes this mindbending is everything needs to be wrapped in a separate typedef with a separate name.  When wrapping primitives that seems completely natural, but when wrapping structs/enums you end up with names like `WrappedUserAccount` that don't feel intuitive to me.  I do want to change this, but I think that needs to be down the line.

FWIW, this is very similar to the actual [code we just shipped for logins](https://github.com/mozilla-mobile/android-components/blob/main/components/service/sync-logins/src/main/java/mozilla/components/service/sync/logins/Types.kt#L16-L28).  On the Rust side `EncryptedLogin` has several nested objects, but on in `android-components` we want it to be flat.  We also need to convert it from the data class that UniFFI defines to the data class defined in the abstract interface (`mozilla.components.concept.storage.EncryptedLogin` )

Reply:
```udl
/// We want to interoperate between `java.util.Timestamp` type and a `chrono`
/// Uniffi can transport a dictionary
dictionary InternalTimestamp {
    i64 seconds;
    i32 nanos;
};

/// Then a toml expression can transform into and from the internal struct
typedef InternalTimestamp Timestamp;
```

> everything needs to be wrapped in a separate typedef with a separate name

Yes, I see that will get tiresome fairly quickly. A quick look at the WebIDL spec: no, there isn't an obvious path forward. Agreed this is an ergonomics issue we shouldn't tackle this minute.

> FWIW, this is very similar to the actual code we just shipped for logins. 

Oh, I'd forgotten about that code. This cleans that up nicely. I figured that an example with Timestamp would show mapping to and from types most obviously out of our control.

Reply:
I'm inclined to agree with this - as above, I think the user wrapping example is confusing enough to us that it will be intractable for people new to uniffi.